pkg/keyidentifier/keyidentifier_test.go

```diff
@@ -30,105 +41,88 @@ func TestIdentifyFiles(t *testing.T) {
 			return nil
 		}
 
-		// This automatic tester can only parse things if they're stored
-		// in a folder hierarchy. So skip the odd test data
-		if strings.HasPrefix(path, "testdata/plaintext") || strings.HasPrefix(path, "testdata/encrypted") {
+		// all json files in testdata/specs are assumed to be valid test specifications
+		if strings.HasSuffix(path, ".json") {
 			testFiles = append(testFiles, path)
 			return nil
 		}
 
 		return nil
 	})
 	require.NoError(t, err, "filepath.Walk")
-
-	for _, path := range testFiles {
-		testIdentifyFile(t, kIdentifer, path)
+	for _, specPath := range testFiles {
+		testIdentifyFile(t, kIdentifier, specPath)
 	}
 }
 
-func testIdentifyFile(t *testing.T, kIdentifer *KeyIdentifier, path string) {
-	pathComponents := strings.Split(path, "/")
-	expected := &KeyInfo{
-		Format: pathComponents[4],
-	}
+func testIdentifyFile(t *testing.T, kIdentifer *KeyIdentifier, specFilePath string) {
+	// load the json file
+	data, err := ioutil.ReadFile(specFilePath)
+	require.NoError(t, err, "reading spec file")
+	var example spec
+	err = json.Unmarshal(data, &example)
```

Comment:
Why change `expected` to `example`?

Reply:
no good reason, mostly because I was piecemeal replacing old code with new, and it was simpler to have a different variable name. You want me to change it back?

Reply:
If you don't feel strong, yes. I think `actual` and `expected` are more clear than `example`

Reply:
makes sense!